pkg/cluster/cluster.go

```diff
@@ -273,10 +273,10 @@ func (c *Cluster) CreateMachine(machine *Machine, i int) error {
 		}
 
 		// Initial provisioning.
-		if err := containerRunShell(name, initScript); err != nil {
+		if err := containerRunShell(name, f(initScript, machine.User())); err != nil {
```

Comment:
This doesn't distinguish between the username and the user's home folder, which are different things. Same on line 279.

Reply:
If you look at the original code, it's hardcoded to be `/root/.ssh/` for the `authorized_keys`.

Once the user is different, say `ubuntu`, this path won't be valid any more to host the `authorized_keys`, that's why we need to make it dynamic since the user could be configurable, by combining the changes made here within the same PR: https://github.com/weaveworks/footloose/blob/master/ssh.go#L44-L50

Reply:
Oh, for the path issue, you're right that I didn't fix that properly and now I've made another push with it fixed.
Thanks for pointing that out.

How to test?

We can create such a user, say `ubuntu`, and make sure it works properly through the `footloose ssh ubuntu@xxx` or `footloose ssh xxx` if the config file has already configured.

Dockerfile:

```
<OMITTED THE REST>

RUN >/etc/machine-id
RUN >/var/lib/dbus/machine-id

# Test with a dedicated user say "ubuntu" for the container
RUN useradd -ms /bin/bash ubuntu

<OMITTED THE REST>
```

_test_user.yaml:

```sh
cluster:
  name: test
  privateKey: cluster-key
machines:
- count: 1
  spec:
    image: brightzheng100/footloose-ubuntu22-ubuntu:arm64
    name: vm-%d
    user: ubuntu               # <--- this does the trick
    networks:
    - footloose-cluster
    portMappings:
    - containerPort: 22
    privileged: true
    volumes:
    - type: volume
      destination: /var
```

And the experience must be the same:
```sh
footloose create -c _test_user.yaml
footloose show -c _test_user.yaml

# by this
footloose ssh vm-0 -c _test_user.yaml
# or this
footloose ssh ubuntu@vm-0 -c _test_user.yaml
```